### PR TITLE
Wayland reconcile, Document how to use cameras and VPN, and mention workarounds

### PR DESCRIPTION
**Pull request by pothos**
This picks the commit from https://github.com/glima/mkosi-intune, reverts to Edge Stable and then I added:

- Document how to use cameras and VPN, and mention workarounds
    
    This documents the changes I needed to get a working setup with Wayland,
    using cameras and the VPN.
    It reconciles the repo with
    https://github.com/glima/mkosi-intune by commenting wayland-only changes
    out and documenting how to enable them again.

### CODE REVIEW

File: mkosi.skeleton/etc/profile.d/mkosi.sh [hunk 1]
```diff
@@ -1,8 +1,11 @@
 export DISPLAY=:0
 export PATH="/opt/microsoft/intune/bin:$PATH"
 export PULSE_SERVER=unix:/run/host_pulse_native
+#WAYLAND: export WAYLAND_DISPLAY=/run/host_wayland_0
+#WAYLAND: export PIPEWIRE_REMOTE=/run/host_pipewire_0
 
-systemctl --user import-environment DISPLAY PATH PULSE_SERVER
+#WAYLAND: alias edge='microsoft-edge-stable --enable-features=UseOzonePlatform --ozone-platform=wayland --enable-features=WebRTCPipeWireCapturer'
+systemctl --user import-environment DISPLAY PATH PULSE_SERVER WAYLAND_DISPLAY PIPEWIRE_REMOTE
```
Comment (pothos) [inline mkosi.skeleton/etc/profile.d/mkosi.sh:8]: (`import-environment` gracefully handles missing variables, this should not harm X11 support)

**Comment by 4nd3r** [comment]
`mkosi` has some cool features coming up in v19, mostly profiles. I was thinking maybe we can leverage that to support both X and Wayland from same repository. But I have to play around with that before.

Also why `seahorse` is required? Never had to use that or do I miss something here?

**Comment by pothos** [comment]
> `mkosi` has some cool features coming up in v19, mostly profiles. I was thinking maybe we can leverage that to support both X and Wayland from same repository. But I have to play around with that before.

Sounds like a good improvement over running `sed` ;)

> Also why `seahorse` is required? Never had to use that or do I miss something here?

Did you change the password before login? I didn't hence I had to use both `passwd` and `seahorse` because the keyring was already created with the old password.

**Comment by pothos** [comment]
Maybe we can use a password prompt in `mkosi.postinst`? Or use `nc -l -p 1234` to get it piped from a second terminal?

**Comment by 4nd3r** [comment]
> Did you change the password before login?

Build image, boot it up, login, passwd.

> Maybe we can use a password prompt in mkosi.postinst?

Let me see...

**Comment by 4nd3r** [comment]
image

Looks like we can ask for password in postinst.

And Intune was happy too.

image

I'll commit that change and move X-related stuff to profiles.

**Comment by 4nd3r** [comment]
@pothos I hacked around a bit and added profiles. Make sure you have latest `mkosi` from the main branch (there might be bugs tho :wink:). In `mkosi.profiles/wl/` you can override files which are different for Wayland.

**Comment by 4nd3r** [comment]
This is probably not relevant anymore due to #14

**Pull request closed without merging**